### ISSUE TITLE
Am I using this wrong?

### ISSUE THREAD

**Issue by FichteFoll** [kaste/SublimeLinter-addon-filter#1]
In my keymap, I have the following bindings:
```js
  // sublime linter
  { "keys": ["ctrl+k", "f"],
    "command": "sublime_linter_addon_cycle_filter_patterns",
    "args": {
      "patterns": ["-annotations:", "annotations:", ""]
    }
  },
  { "keys": ["ctrl+k", "h"], "command": "sublime_linter_toggle_highlights" },
```

Opening the panel and then running the cycle filter patterns command shows a message in the status bar about updating the filter, but the panel's contents don't change. Considering I have two mutually exclusive filters in there, I'd expect at least something to happen.

Here's a video recording of that (Github still does not support videos):
https://x0.at/hyf.mp4

**Comment by kaste** [comment]
I have the following

```
  { "keys": ["ctrl+k", "ctrl+f"],
    "command": "sublime_linter_addon_cycle_filter_patterns",
    "args": {
      "patterns": ["-annotations:", "annotations:", ""]
    }
  },
```

and here's the GIF

![y7kkr5s2cy](https://user-images.githubusercontent.com/8558/45646940-f6192700-bac4-11e8-99f6-6b902cb8f229.gif)

🤔 Need a restart? I cannot be 100% sure if this works after SL updates. I think it should reload the plugin just like every other plugin, but ... 

The code here is relatively simple. It calls a function in SL, then it flashes the status bar. So, it's likely that the function reference I hold here belongs to an old SL instance.

**Comment by FichteFoll** [comment]
This was after a restart already. In fact, restarted again just now and still no change in behavior.

Would be happy to help with debugging if you give me a few hints where to look or which information to provide. I mean, I was the one who requested this feature/addon.

**Comment by kaste** [comment]
The SublimeLinter reloader reloads all plugins, just checked. Can you invoke this command (Command Palette: 'RSL' 'Reload SublimeLinter...'). 

Do you have SL and all plugins installed via PC, or via git clone? Does this affect load order on Sublime startup? Could you check if Sublime gets reloaded on startup before or after the plugins. 

I just assumed SublimeLinter will load/reload before any plugins. That's what I have on my computer (almost every plugin installed via git clone), and that's what the SL reloader does. But if the load order is in fact not predictable, then this plugin would probably have to lazy import SublimeLinter.

**Comment by kaste** [comment]
If it doesn't work, does 

```
import sys
assert sys.modules['SublimeLinter.sublime_linter'].update_buffer_errors == sys.modules['SublimeLinter-addon-filter.plugin'].update_buffer_errors
```

hold?

**Comment by FichteFoll** [comment]
>Do you have SL and all plugins installed via PC, or via git clone? Does this affect load order on Sublime startup? Could you check if Sublime gets reloaded on startup before or after the plugins.

Good thinking there. Yes, I still have a git clone of SL from the time that I worked on it, while I installed the addon via PC. As I don't do that anymore, I should probably go back to the release version. Running the reload command makes it work again.

I suspect that doing it in `plugin_loaded` could still not be enough, however (#2).

**Comment by FichteFoll** [comment]
Hm, even after installing SL from PC as well, I get this load order:
```
reloading plugin SublimeLinter-addon-filter.plugin
reloading plugin SublimeLinter-addon-toggler.package_control_listener
reloading plugin SublimeLinter-addon-toggler.plugin
reloading plugin SublimeLinter-annotations.linter
reloading plugin SublimeLinter-contrib-mypy.linter
reloading plugin SublimeLinter-contrib-rustc.linter
SublimeLinter: WARNING: rustc: Defining 'cls.syntax' has been deprecated. Use http://www.sublimelinter.com/en/stable/linter_settings.html#selector
reloading plugin SublimeLinter-contrib-yamllint.linter
reloading plugin SublimeLinter-flake8.linter
reloading plugin SublimeLinter-jshint.linter
reloading plugin SublimeLinter-lua.linter
reloading plugin SublimeLinter.__init__
reloading plugin SublimeLinter.active_linters_view
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.goto_commands
reloading plugin SublimeLinter.highlight_view
reloading plugin SublimeLinter.log_handler
reloading plugin SublimeLinter.message_view
reloading plugin SublimeLinter.panel_view
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
```

I wonder whether ST sorts installed packages differently from user packages?

```
~/.config/sublime-text-3 ↻ ls *Packages/SublimeLinter* -d
Installed Packages/SublimeLinter-addon-filter.sublime-package      Installed Packages/SublimeLinter-jshint.sublime-package  Packages/SublimeLinter-flake8
Installed Packages/SublimeLinter-addon-toggler.sublime-package     Installed Packages/SublimeLinter-lua.sublime-package
Installed Packages/SublimeLinter-annotations.sublime-package       Installed Packages/SublimeLinter.sublime-package
Installed Packages/SublimeLinter-contrib-mypy.sublime-package      Packages/SublimeLinter-annotations
Installed Packages/SublimeLinter-contrib-rustc.sublime-package     Packages/SublimeLinter-contrib-mypy
Installed Packages/SublimeLinter-contrib-yamllint.sublime-package  Packages/SublimeLinter-contrib-rustc
Installed Packages/SublimeLinter-flake8.sublime-package            Packages/SublimeLinter-contrib-sublime-syntax
```

Edit: Well, it kind of makes sense …
```py
>>> ord(".")
46
>>> ord("-")
45
```

**Comment by kaste** [comment]
LOL, funny stuff.

On my computer, Sublime first loads the 'Installed packages' then the ones in 'Packages'. But, the `plugin_loaded` callback runs *after* Sublime logs `plugins loaded`. So, I *think* we're just fine doing it lazy in `plugins_loaded`.

**Comment by FichteFoll** [comment]
Yeah, I think it should be safe for now to assume that `plugin_loaded` is called after the first initial scan for plugins, although it isn't documented anywhere, so hard to rely on.

**Comment by kaste** [comment]
I merged and pushed #1. 🤞
